core/domain/platform_parameter_domain.py

```diff
@@ -35,3 +61,976 @@ class PlatformParameterChange(change_domain.BaseChange):
+                'received %s.' % (
+                    feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION,
+                    schema_version))
+
+        return cls(
+            [
+                PlatformParameterFilter.from_dict(filter_dict)
+                for filter_dict in rule_dict['filters']],
+            rule_dict['value_when_matched'],
+        )
+
+
+class PlatformParameterMetadata(python_utils.OBJECT):
+    """Domain object for metadatas of platform parameters."""
+
+    def __init__(self, is_feature, stage):
+        self._is_feature = is_feature
+        self._stage = stage
+
+    @property
+    def is_feature(self):
+        """Returns True if the corresponding platform parameter is a feature
+        flag.
+
+        Returns:
+            bool. True if the corresponding platform parameter is a feature
+            flag.
+        """
+        return self._is_feature
+
+    @property
+    def stage(self):
+        """Returns the stage of the feature flag.
+
+        Returns:
+            str. The stage of the feature flag.
+        """
+        return self._stage
+
+    def to_dict(self):
+        """Returns a dict representation of the PlatformParameterMetadata
+        domain object.
+
+        Returns:
+            dict. A dict mapping of all fields of PlatformParameterMetadata
+            object.
+        """
+        return {
+            'is_feature': self._is_feature,
+            'stage': self.stage,
+        }
+
+    @classmethod
+    def from_dict(cls, metadata_dict):
+        """Returns an PlatformParameterMetadata object from a dict.
+
+        Args:
+            metadata_dict: dict. A dict mapping of all fields of
+                PlatformParameterMetadata object.
+
+        Returns:
+            PlatformParameterMetadata. The corresponding
+            PlatformParameterMetadata domain object.
+        """
+        return cls(
+            metadata_dict['is_feature'],
+            metadata_dict['stage'],
+        )
+
+
+class PlatformParameter(python_utils.OBJECT):
+    """Domain object for platform parameters."""
+
+    CMD_CHANGE_VARIABLE_SETTING = 'change_variable_setting'
+
+    DATA_TYPE_PREDICATES_DICT = {
+        'bool': lambda x: isinstance(x, bool),
+        'string': lambda x: isinstance(x, python_utils.BASESTRING),
+        'number': lambda x: isinstance(x, (float, int)),
+    }
+
+
+    PARAMETER_NAME_REGEXP = r'^[A-Za-z0-9_]{1,100}$'
+
+    def __init__(
+            self, name, description, data_type, rules,
+            rule_schema_version, metadata):
+        if re.match(self.PARAMETER_NAME_REGEXP, name) is None:
+            raise Exception(
+                'Invalid parameter name, expected to match regexp %s, '
+                'got %s.' % (
+                    self.PARAMETER_NAME_REGEXP, name))
+
+        self._name = name
+        self._description = description
+        self._data_type = data_type
+        self._rules = rules
+        self._rule_schema_version = rule_schema_version
```

Comment:
Not sure if I understand this question.
The schema version in PlatformParameter objects represents the schema of rule dicts, which is then used in the creation of rule domain objects for schema check -- if the schema version is not latest and there isn't code to convert older schema to the latest one, an Exception is raised.
This aims at cases where the dict stored in the database is not in latest schema, with the schema version (which is also stored in database) we can tell whether the dict is valid.

Reply:
Sorry my question wasn't clear--I was wondering why the schema is replicated both here & in the rule class above. Why not perform the schema validation in this class's validate method, instead?

Reply:
Refactored the code -- now only `PlatformParameter` has the schema version field, which will be validated during creation in `PlatformParameter.fromDict()`
I didn't put the schema version check to its `validate` method because, I think it's necessary to make sure the rule dict schema is up-to-date before creating rule objects, -- if we defer the checks to the validation, there might be unexpected exceptions from creating rule objects using the old schema.

Reply:
That makes sense--thanks!